### ISSUE TITLE
Extend require-commit-hash (#436) to nested uses: inside composite actions

### ISSUE THREAD

**Issue by thehcma** [rhysd/actionlint#738]
## Summary

Please consider extending the opt-in SHA-pin check from #436 / #198 so it also flags **non-SHA `uses:` refs inside composite actions** (`action.yml` / `action.yaml`), including when a workflow calls a **local** composite via `uses: ./…`.

**This is not a duplicate of #436.** #436 (and the related asks in #435 / #198) cover **direct** workflow-level `uses:`. That alone would **not** catch the failure mode below: a workflow can pin `owner/helpers/actions/setup-foo@<40-char-sha>` and still pull a mutable tag from a nested step inside that composite.

Related:
- #198 (security best-practice options)
- #435 (warn on `@main` / `@master`)
- #436 (optional `require-commit-hash` for workflow-level `uses:`)

## Not covered by #436

From #436's approach (`require-commit-hash` on repo-action specs in the workflow), this still passes:

```yaml
# .github/workflows/ci.yml
- uses: ./actions/setup-foo   # or owner/helpers/actions/setup-foo@<full-sha>
```

…while this nested step inside the composite is what actually runs (and what GitHub's `sha_pinning_required` evaluates against the **caller**):

```yaml
# actions/setup-foo/action.yml
runs:
  using: composite
  steps:
    - uses: actions/cache@v6.1.0   # mutable tag - invisible to a workflow-only pin check
```

## Security rationale

GitHub's own hardening guidance is to pin third-party actions to a full-length commit SHA so a compromised or retagged ref cannot silently change what runs - see [Using third-party actions](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) in [Security hardening for GitHub Actions](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions).

Repos can also enable **`sha_pinning_required`** on Actions permissions. When that is on, GitHub evaluates **nested** `uses:` inside composite actions against the **calling repository's** policy - not only the top-level workflow `uses:` lines.

Consequence:

1. A consumer workflow pins `owner/helpers/actions/setup-foo@<40-char-sha>` (looks fine; satisfies #436).
2. That composite still contains e.g. `uses: actions/cache@v6.1.0`.
3. With `sha_pinning_required`, CI fails at job setup with:
   `The action actions/cache@… is not allowed … must be pinned to a full-length commit SHA`
4. Without that setting, the same nested tag is a **supply-chain hole**: the consumer believed they pinned everything, but a mutable tag inside the composite can still move.

Mutable tags inside composites therefore defeat both (a) the security invariant of SHA pinning and (b) GitHub's enforcement setting, while remaining invisible to a workflow-only pin check.

Real-world shape we hit: an org composite that wraps Corepack + `actions/cache`. Consumers SHA-pinned the composite; nested `actions/cache@vX.Y.Z` still broke (or would silently weaken) callers that turned on `sha_pinning_required`.

## Why actionlint is a good place (for the local case)

actionlint already understands local composite metadata when workflows use `./path` (`checkLocalCompositeActionRuns`, input validation, etc.). Extending #436's rule to "every third-party nested `uses:` in this local `action.yml` must be `@[0-9a-f]{40}`" can be done **offline** with no GitHub API - consistent with actionlint's design.

Checking **remote** composites transitively needs network/auth (see discussion on #198 / third-party pin scanners). That can stay out of scope; the local + "when analyzing `action.yml` directly" case alone would catch org-owned composites before they ship.

## Proposed behavior (sketch)

Opt-in under the same config family as #436's `require-commit-hash` (either fold nested checks into that flag, or a sibling such as `require-commit-hash-in-composites`):

1. When linting workflows: for each `uses: ./…` composite, parse that `action.yml` and report nested third-party `uses:` that are not full-length SHAs.
2. When/if actionlint gains first-class linting of `action.yml` files: apply the same rule there (today pointing actionlint at `action.yml` treats it as a workflow and errors on missing `on:` / `jobs:`).
3. Skip `./` and `docker://` nested refs (or require digests for Docker - optional).
4. Allow `# vX.Y.Z` comments after the SHA (common pinning style).

Example that should fail under the option:

```yaml
# actions/setup-foo/action.yml
runs:
  using: composite
  steps:
    - uses: actions/cache@v6.1.0   # mutable tag
```

Example that should pass:

```yaml
    - uses: actions/cache@55cc8345863c7cc4c66a329aec7e433d2d1c52a9 # v6.1.0
```

## Alternatives considered

- **zizmor** `unpinned-uses` and tools like `check-actions-sha-pinning` already cover much of the security-scanner niche (noted on #435). Still valuable for actionlint users who want one offline workflow+local-action gate in CI without adopting a second tool - especially for private org composites.
- Org-specific linters can special-case one composite; a general nested-`uses:` rule scales to any local composite.

Happy to refine the config shape to match whatever lands for #436.
